contracts/cryptoblades.sol

```diff
@@ -517,65 +522,54 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         _updateCharacterMintFee();
     }
 
-    function mintWeaponN(uint32 num, uint8 chosenElement, uint256 eventId)
-        external
-        onlyNonContract
-        oncePerBlock(msg.sender)
-    {
-        uint8 chosenElementFee = chosenElement == 100 ? 1 : 2;
-        _payContractConvertedSupportingStaked(msg.sender, usdToSkill(getMintWeaponFee() * num * chosenElementFee));
-        _mintWeaponNLogic(num, chosenElement, eventId);
-    }
-
-    function mintWeapon(uint8 chosenElement, uint256 eventId) external onlyNonContract oncePerBlock(msg.sender) {
+    function generateWeaponSeed(uint32 quantity, uint8 chosenElement, uint256 eventId) external onlyNonContract oncePerBlock(msg.sender) {
+        require(quantity > 0 && quantity <= 10);
```

Comment:
adding
userVars[msg.sender][USERVAR_CLAIM_WEAPON_DATA] == 0
to the require statement here and in the other generate statement would be user friendly, to prevent accidental loss of funds. But in any case this is fine.